rs/node_rewards/canister/src/main.rs

```diff
@@ -42,26 +44,14 @@ fn post_upgrade() {
     schedule_timers();
 }
 
-fn schedule_timers() {
-    schedule_registry_sync();
-}
-
 // The frequency of regular registry syncs.  This is set to 1 hour to avoid
 // making too many requests.  Before meaningful calculations are made, however, the
 // registry data should be updated.
-const REGISTRY_SYNC_INTERVAL_SECONDS: Duration = Duration::from_secs(60 * 60); // 1 hour
+const SYNC_INTERVAL_SECONDS: Duration = Duration::from_secs(60 * 60); // 1 hour
 
-fn schedule_registry_sync() {
-    ic_cdk_timers::set_timer_interval(REGISTRY_SYNC_INTERVAL_SECONDS, move || {
-        spawn(async move {
-            let store = REGISTRY_STORE.with(|s| s.clone());
-            // panicking here is okay because we are using an interval instead of a timer that
-            // has to reschedule itself.
-            store
-                .sync_registry_stored()
-                .await
-                .expect("Could not sync registry store!");
-        });
+fn schedule_timers() {
+    ic_cdk_timers::set_timer_interval(SYNC_INTERVAL_SECONDS, move || {
+        spawn(NodeRewardsCanister::sync_all(&CANISTER));
```

Comment:
I'm not sure if this is good or bad, but I think it's generally better to separate out different recurring tasks into their own thing so that if frequencies ever don't match up, it's very easy to change them independently.